src/dev/flang/be/jvm/runtime/Runtime.java

```diff
@@ -745,15 +742,15 @@ public static byte[] fuzion_sys_env_vars_get0(Object d)
    * Creates a new instance of String from the byte array passed as argument,
    * assuming the byte array contains an UTF-8 encoded string.
    *
-   * @param b byte array consisting of a string encoded as UTF-8 bytes
+   * @param b byte array consisting of a string encoded as UTF-8 bytes (0-terminated)
    *
    * @return the string from the array, as an instance of Java's String
    */
   public static String fuzion_java_string_to_java_object0(byte[] b)
   {
     unsafeIntrinsic();
 
-    return new String(b, StandardCharsets.UTF_8);
+    return new String(b, 0, b.length -1, StandardCharsets.UTF_8);
```

Comment:
I would prefer not to add the overhead of the extra byte array creation in the JVM backend. If it is too hard to avoid this here, at least we shoudl add a `NYI: OPTIMZATION` comment.